server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java

```diff
@@ -102,6 +103,18 @@ protected void sendReplicaRequest(
         }
     }
 
+    @Override
+    protected ClusterBlockLevel globalBlockLevel() {
+        // resync should never be blocked
```

Comment:
maybe add here "because it's an internal action"

Reply:
Sure.